File: Missing-in-Array.py
class Solution:
    def missingNum(self, arr):
        # code here
        n=len(arr)
        # a=n+1 
        # sum=a*(a+1)//2
        # s1=0
        # for i in range(n):
        #     s1+=arr[i]
        # return sum-s1
        
        xor1=0
        xor2=0
        for i in range(1,n+2):
            xor1^=i
        for i in arr:
            xor2^=i
        return xor1^xor2

File: Missing-in-Array.js
/**
 * @param {number[]} arr
 * @returns {number}
 */
class Solution {
    missingNum(arr) {
        // code here
        let n=arr.length
        // let a=n+1
        // let sum= a*(a+1)/2
        // let s1=0
        // for (let i=0; i<n;i++){
        //     s1=s1+arr[i]
        // }
        // return sum-s1
        
        let xor1=0
        let xor2=0
        for (let i=1;i<n+2;i++){
            xor1=xor1^i
        } 
        for (let i=0;i<n;i++){
            xor2=xor2^arr[i]
        } 
        return xor1^xor2
    }
}
